flood_adapt/object_model/site.py

```diff
@@ -1,88 +1,153 @@
-            "datum_overland_model",
-            "diff_datum_offshore_overland",
-            "tidal_components",
-            "ambient_air_pressure",
-            "floodmap_no_data_value",
-            "floodmap_units",
-        ]
-        mandatory_slr = ["vertical_offset", "relative_to_year"]
-        mandatory_risk = ["return_periods", "flooding_threshold"]
-        mandatory_gui = ["tide_harmonic_amplitude"]
-        mandatory_dem = ["filename", "units", "indexfilename"]
-        mandatory_fiat = [
-            "exposure_crs",
-            "aggregation_shapefiles",
-            "aggregation_field_names",
-            "floodmap_type",
-        ]
-        if val.validate_content_config_file(
-            config, self.config_path, mandatory_attributes
-        ):
-            for attr in mandatory_attributes:
-                # Check if attributes have correct dictionary keys (if applicable)
-                if attr == "sfincs":
-                    val.validate_content_config_file(
-                        config[attr], self.config_path, mandatory_sfincs
-                    )
-                elif attr == "slr":
-                    val.validate_content_config_file(
-                        config[attr], self.config_path, mandatory_slr
-                    )
-                elif attr == "risk":
-                    val.validate_content_config_file(
-                        config[attr], self.config_path, mandatory_risk
-                    )
-                elif attr == "gui":
-                    val.validate_content_config_file(
-                        config[attr], self.config_path, mandatory_gui
-                    )
-                elif attr == "dem":
-                    val.validate_content_config_file(
-                        config[attr], self.config_path, mandatory_dem
-                    )
-                elif attr == "fiat":
-                    val.validate_content_config_file(
-                        config[attr], self.config_path, mandatory_fiat
-                    )
-                # Add the attributes
-                setattr(self, attr, config[attr])
-
-            for attr in non_mandatory_attributes:
-                for key in config:
-                    if attr == key:
-                        setattr(self, attr, config[attr])
+    """Class for general variables of the object_model"""
+
+    model: SiteConfigModel
+
+    @staticmethod
+    def load_file(filepath: Path):
+        """create SiteConfig from toml file"""
+
+        obj = SiteConfig()
+        with open(filepath, mode="rb") as fp:
+            toml = tomli.load(fp)
+        obj.model = SiteConfigModel.parse_obj(toml)
+        return obj
+
+    @staticmethod
+    def load_dict(data: dict):
+        """create Synthetic from object, e.g. when initialized from GUI"""
+
+        obj = SiteConfig()
+        obj.model = SiteConfig.parse_obj(data)
+        for key, value in obj.model.dict().items():
+            setattr(obj, key, value)
```

Comment:
please delete, we decided we leave the additional layer "attrs", it was too hard otherwise